lib/iris/fileformats/__init__.py

```diff
@@ -27,13 +27,11 @@
                                    UriProtocol, LeadingLine)
 from . import abf
 from . import um
+
 try:
-    import iris_grib as igrib
+    from . import grib as igrib
 except ImportError:
-    try:
-        from . import grib as igrib
-    except ImportError:
-        igrib = None
+    igrib = None
```

Comment:
at present the grib module is dependent on gribapi which may not be available, so a conditional import is used
e.g. windows, python3.x

adopting ECCodes may help this, but we may still maintain a conditional import and keep GRIB as an optional dependency

either way, for now it is required